### PR TITLE
Directly: Only initialize widget for users who will use it

### PR DESCRIPTION
**Pull request by mattwondra**
Currently the Directly RTM widget initializes for all users who visit `/help/contact`. This PR ensures that only users who are meant to see the widget will have the third-party code initialized.

### To test

Check for initialization by opening Network dev tools and filtering on the term `directly`. If you see any assets come through, initialization has occurred.

#### 1. Paid users should not see initialization
Sign in as a user with paid upgrades. Visit http://calypso.localhost:3000/help/contact. You should not see initialization happen.

#### 2. Unpaid English-speaking users should see initialization
Sign in as a user with no paid upgrades and account language set to English. Visit http://calypso.localhost:3000/help/contact. You should see initialization assets come across the network.

#### 3. Unpaid non-English speakers should not see initialization
On your unpaid account, set your language to something other than `en`. Visit http://calypso.localhost:3000/help/contact. You should not see initialization happen.

#### 4. Unpaid non-English speakers whose language switches to English should see initialization
On your unpaid account, set your language to something other than `en`. Visit http://calypso.localhost:3000/help/contact. You should not see initialization happen.

Now we'll send an action through Redux Dev Tools to change the language settings on the fly.

- Open Redux Dev Tools
- Find the `USER_RECEIVE` action and click it (should be one of the first dispatched)
- Open the "Raw" tab and copy the entire action
- Click the "keyboard" icon to dispatch a new action
- Paste in the copied action
- Find the `localeSlug` property and change the value to `'en'`
- Click the Dispatch button

You should now see the form change and Directly initialization assets come through the network.

### CODE REVIEW

File: client/me/help/help-contact/index.jsx [hunk 1]
```diff
@@ -518,7 +540,7 @@ const HelpContact = React.createClass( {
 					showSubjectField: false,
 					showHowCanWeHelpField: false,
 					showHowYouFeelField: false,
-					showSiteField: hasMoreThanOneSite,
+					showSiteField: false,
```
Comment (dllh) [inline client/me/help/help-contact/index.jsx]: Is this change intentional? Seems like we're updating Kayako functionality here and maybe introducing a regression.

Comment (mattwondra) [inline client/me/help/help-contact/index.jsx]: @dllh Due to rebases and code merges this morning this PR isn't reading clean. This change is already merged into master and this branch needs a rebase, which I'll do in just a moment.

**Comment by jordwest** [comment]
- [x] Paid users should not see initialization
- [x] Unpaid English-speaking users should see initialization
- [x] Unpaid non-English speakers should not see initialization
 - [x]  Unpaid non-English speakers whose language switches to English should see initialization

Tested and the code looks good to me 👍 

In case anyone else comes sees the same thing:
Right after switching to `es` then navigating to /help/contact, Directly initialized (though it did show the note -- in Spanish -- that support is only available in English). However after refreshing, it reverted to Forums. I couldn't reproduce this more than once, and I've seen a similar thing happen before in other parts of Calypso. It's as if the old `localeSlug` sometimes lingers in the Redux store after a switch, but not reproducibly. Not a big deal and not related to this PR, but thought its worth mentioning.

**Comment by mattwondra** [comment]
@jordwest Thanks so much for your time reviewing. Sorry to say that I realized today this PR has conflicting logic with another open PR (https://github.com/Automattic/wp-calypso/pull/11768). So I'm closing this out in favor of another approach here: https://github.com/Automattic/wp-calypso/pull/11871

The core idea is the same, except some of the complicated logic in the component has been broken out into a helper. There's more details and background on the PR.
